resources/testdriver.js

```diff
@@ -769,6 +769,33 @@
                     },
                 }
             },
+            speculation: {
```

Comment:
Please add jsdoc, as it is used to generate up-to-date https://web-platform-tests.org/writing-tests/testdriver.html. You can refer to other methods.